### PR TITLE
Added SkipIntrospectionFields option on MaxDepthAnalyzer

### PR DESCRIPTION
**Pull request by michaelstaib**

### CODE REVIEW

File: src/HotChocolate/Core/test/Validation.Tests/MaxDepthRuleTestsSkipIntrospection.cs [hunk 1]
```diff
@@ -0,0 +1,186 @@
+﻿using Microsoft.Extensions.DependencyInjection;
```
Comment (PascalSenn) [inline src/HotChocolate/Core/test/Validation.Tests/MaxDepthRuleTestsSkipIntrospection.cs:1]: I think there are some snapshots missing for this test class

File: src/HotChocolate/Core/src/Validation/Rules/MaxExecutionDepthVisitor.cs [hunk 2]
```diff
@@ -42,6 +42,12 @@ public MaxExecutionDepthVisitor(IMaxExecutionDepthOptionsAccessor options)
         FieldNode node,
         IDocumentValidatorContext context)
     {
+        if (_options.SkipIntrospectionFields &&
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Validation/Rules/MaxExecutionDepthVisitor.cs:45]: This also skips the __typename field. Is this intended?

Comment (michaelstaib) [reply src/HotChocolate/Core/src/Validation/Rules/MaxExecutionDepthVisitor.cs:45]: yes its intended

Comment (michaelstaib) [reply src/HotChocolate/Core/src/Validation/Rules/MaxExecutionDepthVisitor.cs:45]: All introspection

File: src/HotChocolate/Core/test/Validation.Tests/MaxDepthRuleTestsSkipIntrospection.cs (hunk 1 above)
Comment (michaelstaib) [reply src/HotChocolate/Core/test/Validation.Tests/MaxDepthRuleTestsSkipIntrospection.cs:1]: done

**Pull request merged**